### PR TITLE
Fix -0 input for numerical fields

### LINKED ISSUE #132: [PE-D] Able to add a '-0' to the numerical field of leaves
-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: marcuschj/ped#3

### PR DESCRIPTION
**Pull request by 1waykiat**
Added/Modified methods in StringUtil to check for negative signs.
Added test cases for -0 inputs as well.

Resolves #132 and #135

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -60,7 +60,20 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
             int value = Integer.parseInt(s);
-            return value > 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
+            // "+1" is successfully parsed by Integer#parseInt(String)
+            return value > 0 && !s.startsWith("+") && !s.contains("-");
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+    }
+
+    public static boolean isNonNegativeInteger(String s) {
```
Comment (szelongq) [inline src/main/java/seedu/address/commons/util/StringUtil.java:70]: From the checkstyle error messages, I think maybe you should add a javadoc comment here?

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -60,7 +60,20 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            // "+1" is successfully parsed by Integer#parseInt(String)
+            return value >= 0 && !s.startsWith("+") && !s.contains("-");
```
Comment (szelongq) [inline src/main/java/seedu/address/commons/util/StringUtil.java:76]: Also apart from adding the appropriate tests for hourlySalary and the other attributes using this method, maybe you could add tests for this method by itself? I think it would also be another chance to show that you know how to make test cases.

**Review by szelongq — COMMENTED** [review]
LGTM! Just some checkstyle errors to rectify, and I also made a small suggestion if you're ok with it otherwise it's fine!

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 3]
```diff
@@ -77,6 +80,44 @@ public void isNonNegativeDoubleWithTwoOrLessDecimal() {
...
+
+        // EP: numbers with white space
+        assertFalse(StringUtil.isNonNegativeInteger(" 10 ")); // Leading/trailing spaces
+        assertFalse(StringUtil.isNonNegativeInteger("1 0")); // Spaces in the middle
+
+        // EP: number larger than Integer.MAX_VALUE
+        assertFalse(StringUtil.isNonNegativeInteger(Long.toString(Integer.MAX_VALUE + 1)));
+
+        // EP: valid numbers, should return true
+        assertTrue(StringUtil.isNonNegativeInteger("1")); // Boundary value
```
Comment (szelongq) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:116]: Just a small nitpick, but I think 1 is not a boundary value? Since this is for non-negative integers, so the boundary values should be -1 and 0?

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 4]
```diff
@@ -26,6 +26,9 @@ public void isNonZeroUnsignedInteger() {
         // EP: zero
         assertFalse(StringUtil.isNonZeroUnsignedInteger("0"));
 
+        // EP: negative zero
+        assertFalse(StringUtil.isNonNegativeInteger("-0"));
```
Comment (szelongq) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:30]: Maybe this should be isNonZeroUnsignedInteger instead since this is in its test case?

**Review by szelongq — COMMENTED** [review]
Just a few more small nitpicks, otherwise LGTM!

**Review by szelongq — APPROVED** [review]
The previous nitpicks were really minor so I fixed it first to accommodate changes to other PRs, otherwise as mentioned earlier it LGTM!
